system-parachains/asset-hubs/asset-hub-polkadot/tests/tests.rs

```diff
@@ -954,3 +954,291 @@ fn governance_authorize_upgrade_works() {
+			assert_relative_eq!(
+				projected_total_issuance as f64,
+				(2_700_000_000 * UNITS) as f64,
+				max_relative = 0.001
+			);
+		});
+	}
+
+	// Sets the view of the relay chain block number.
+	fn set_relay_number(n: RC_BlockNumber) {
+		ValidationData::<Runtime>::set(
+			Some( PersistedValidationData {
+				parent_head: vec![].into(),
+				relay_parent_number: n,
+				max_pov_size: Default::default(),
+				relay_parent_storage_root: Default::default(),
+			})
+		);
+	}
+
+	const TWO_YEAR_STEP_RATE: f64 = 0.2628;
+	const TARGET_TI: u128 = 2_100_000_000;
+	// The extrapolated TI on March 14th, 2026. Number deviates slightly from the community's initial extrapolation of
+	// 1_676_733_867 used in [Ref 1710](https://polkadot.subsquare.io/referenda/1710).
+	const MARCH_TI: u128 = 1_676_562_737;
+	const MARCH_14_2026: RC_BlockNumber = 30_367_108;
+
+	// The transition from set emission to stepped emission works.
+	#[test]
+	fn set_to_stepped_inflation_transition_works() {
+		ExtBuilder::<Runtime>::default()
+		.build()
+		.execute_with(|| {
+			// Check before transition date.
+			let (to_stakers, to_treasury) = <staking::EraPayout as EraPayout<Balance>>::era_payout(
+				123, // ignored
+				456, // ignored
+				MILLISECONDS_PER_DAY,
+			);
+			assert_relative_eq!(to_stakers as f64, (279_477 * UNITS) as f64, max_relative = 0.001);
+			assert_relative_eq!(to_treasury as f64, (49_320 * UNITS) as f64, max_relative = 0.001);
+			assert_relative_eq!(
+				(to_stakers as f64 + to_treasury as f64),
+				(328_797 * UNITS) as f64,
+				max_relative = 0.001
+			);
+
+			// Check after transition date.
+			set_relay_number(MARCH_14_2026);
+			let (to_stakers, to_treasury) = <staking::EraPayout as EraPayout<Balance>>::era_payout(
+				123, // ignored
+				456, // ignored
+				MILLISECONDS_PER_DAY,
+			);
+			let era_rate = Perbill::from_float(TWO_YEAR_STEP_RATE) * Perbill::from_float(0.5) * Perbill::from_float(1.0/365.25);
+			let assumed_payout = era_rate * (TARGET_TI - MARCH_TI);
+			assert_relative_eq!(
+				(to_stakers as f64 + to_treasury as f64),
+				(assumed_payout * UNITS) as f64,
+				max_relative = 0.00001
+			);
+		});
+	}
+
+	// The emission values for the two year periods are as expected.
+	#[test]
+	fn stepped_inflation_two_year_values_correct() {
+		ExtBuilder::<Runtime>::default()
+		.build()
+		.execute_with(|| {
+			let two_years: RC_BlockNumber = RC_YEARS * 2;
+
+			// First period - March 14, 2026 -> March 14, 2028.
+			set_relay_number(MARCH_14_2026);
+			let (to_stakers, to_treasury) = <staking::EraPayout as EraPayout<Balance>>::era_payout(
+				123, // ignored
+				456, // ignored
+				MILLISECONDS_PER_DAY,
+			);
+			let two_year_rate = Perbill::from_float(TWO_YEAR_STEP_RATE);
+			let first_period_emission = two_year_rate * (TARGET_TI - MARCH_TI);
+			assert_relative_eq!(
+				(to_stakers as f64 + to_treasury as f64) * 365.25 * 2.0,
+				(first_period_emission * UNITS) as f64,
+				max_relative = 0.00001
+			);
```

Comment:
```suggestion
			);
			assert_relative_eq!((first_period_emission * UNITS) as f64, 123....); 
```
Some sanity check values please for human verification. Just to check that the order of magnitude looks good.